### PR TITLE
Fix TableIdentsExtractor to be able to extract from AliasSymbol

### LINKED ISSUE #13709: `IllegalStateException[Symbol 'io.crate.expression.symbol.Symbol' not supported]` when trying to select on ambiguous column name
### CrateDB version

5.2.2

### CrateDB setup information

only reproducible on multi-node cluster (!)

### Problem description

would consider this rather as a ux improvement, not a high priority :bug:

### Steps to Reproduce

```sql
CREATE TABLE t_symbol (b real); 
WITH base AS (SELECT b as b, b as b FROM t_symbol) 
    SELECT  b  FROM base LIMIT 1;
```

### Actual Result

(on multi-node cluster)
```
Error!

IllegalStateException[Symbol 'io.crate.expression.symbol.Symbol' not supported]

java.lang.IllegalStateException: Symbol 'io.crate.expression.symbol.Symbol' not supported
	at io.crate.analyze.TableIdentsExtractor$TableIdentSymbolVisitor.visitSymbol(TableIdentsExtractor.java:83)
	at io.crate.analyze.TableIdentsExtractor$TableIdentSymbolVisitor.visitSymbol(TableIdentsExtractor.java:79)
	at io.crate.expression.symbol.SymbolVisitor.visitAlias(SymbolVisitor.java:82)
	at io.crate.expression.symbol.AliasSymbol.accept(AliasSymbol.java:71)
	at io.crate.analyze.TableIdentsExtractor.extract(TableIdentsExtractor.java:76)
	at io.crate.exceptions.AmbiguousColumnException.getTableIdents(AmbiguousColumnException.java:42)
	at io.crate.auth.AccessControlImpl$MaskSensitiveExceptions.visitTableScopeException(AccessControlImpl.java:896)
	at io.crate.auth.AccessControlImpl$MaskSensitiveExceptions.visitTableScopeException(AccessControlImpl.java:884)
	at io.crate.exceptions.TableScopeException.accept(TableScopeException.java:34)
	at io.crate.auth.AccessControlImpl.ensureMaySee(AccessControlImpl.java:148)
```

### Expected Result

(as on a single node)
```
Error!

AmbiguousColumnException[Column "b" is ambiguous]
```

### PR DESCRIPTION
**Pull request by jeeminso**
## Summary of the changes / Why this improves CrateDB
Fixes #13709

## Checklist

 - [x] Added an entry in `CHANGES.txt` for user facing changes
 - [x] Updated documentation & `sql_features` table for user facing changes
 - [x] Touched code is covered by tests
 - [x] [CLA](https://crate.io/community/contribute/cla/) is signed
 - [x] This does not contain breaking changes, or if it does:
    - It is released within a major release
    - It is recorded in ``CHANGES.txt``
    - It was marked as deprecated in an earlier release if possible
    - You've thought about the consequences and other components are adapted
      (E.g. AdminUI)

### CODE REVIEW

File: server/src/main/java/io/crate/analyze/TableIdentsExtractor.java [hunk 1]
```diff
@@ -145,6 +146,11 @@ public Collection<RelationName> visitParameterSymbol(ParameterSymbol parameterSy
         public Collection<RelationName> visitSelectSymbol(SelectSymbol selectSymbol, Void context) {
             return selectSymbol.relation().accept(RELATION_TABLE_IDENT_EXTRACTOR, context);
         }
+
+        @Override
+        public Collection<RelationName> visitAlias(AliasSymbol aliasSymbol, Void context) {
+            return aliasSymbol.symbol().accept(this, context);
+        }
```
Comment (jeeminso) [inline server/src/main/java/io/crate/analyze/TableIdentsExtractor.java:153]: It was just forgotten to override `aliasSymbol`. Do we need tests/changes doc entry?

Comment (mfussenegger) [reply server/src/main/java/io/crate/analyze/TableIdentsExtractor.java:153]: Yes, anything that gets backported and has a fixes entry should have a test case.

File: server/src/main/java/io/crate/analyze/TableIdentsExtractor.java [hunk 2]
```diff
@@ -145,6 +146,11 @@ public Collection<RelationName> visitParameterSymbol(ParameterSymbol parameterSy
         public Collection<RelationName> visitSelectSymbol(SelectSymbol selectSymbol, Void context) {
             return selectSymbol.relation().accept(RELATION_TABLE_IDENT_EXTRACTOR, context);
         }
+
+        @Override
+        public Collection<RelationName> visitAlias(AliasSymbol aliasSymbol, Void context) {
```
Comment (mfussenegger) [inline server/src/main/java/io/crate/analyze/TableIdentsExtractor.java:151]: Would it make sense to use `DefaultTraversalSymbolVisitor` for this visitor? Then this override afaik wouldn't be necessary.

Comment (jeeminso) [reply server/src/main/java/io/crate/analyze/TableIdentsExtractor.java]: Thanks made the change but it required some more changes.

File: server/src/main/java/io/crate/analyze/TableIdentsExtractor.java [hunk 3]
```diff
@@ -73,110 +67,115 @@ public static Collection<RelationName> extract(Iterable<? extends Symbol> symbol
...
-            return List.of(field.relation());
+        public Void visitField(ScopedSymbol field, Collection<RelationName> context) {
+            context.add(field.relation());
+            return null;
         }
 
         @Override
-        public Collection<RelationName> visitMatchPredicate(MatchPredicate matchPredicate, Void context) {
-            Set<RelationName> relationNames = new HashSet<>();
+        public Void visitMatchPredicate(MatchPredicate matchPredicate, Collection<RelationName> context) {
```
Comment (mfussenegger) [inline server/src/main/java/io/crate/analyze/TableIdentsExtractor.java:122]: I think this matches the default implementation and could be removed

**Review by mfussenegger — APPROVED** [review]

File: server/src/main/java/io/crate/analyze/TableIdentsExtractor.java (hunk 3 above)
Comment (jeeminso) [reply server/src/main/java/io/crate/analyze/TableIdentsExtractor.java:122]: Thank you, updated it.

**Pull request merged**